test/Frontend/skip-function-bodies.swift

```diff
@@ -39,6 +39,21 @@
 @inline(never)
 func _blackHole(_ s: String) {}
 
+struct StructOuter {
+  @usableFromInline
+  struct StructInner {}
+}
+extension StructOuter.StructInner {
+  @inlinable
+  func inlinableInnerFunc() {
+    let ALLNOTYPECHECK_innerStructAndFuncLocal = 1
+    _blackHole("@inlinable func in inner @usableFromInline struct")
+    // CHECK-NONINLINE-SIL: "@inlinable func in inner @usableFromInline struct"
+    // CHECK-NONINLINE-TEXTUAL-NOT: "@inlinable func in inner @usableFromInline struct"
```

Comment:
I think I see the inconsistency in the compiler. This isn't really inlinable, even with `StructInner` being `@usableFromInline` if `StructOuter` is internal none of this can be inlined from outside of the module.

If this is also the case in the original bug, we probably don't want to generate the SIL for it. We should probably diagnose it too.